### PR TITLE
Add risk analysis and pre-mortems content and media

### PR DESCRIPTION
**Pull request by AndrewJanuary**
Firstly apologies that I've had to fork the repo first. Looks like I'm lacking write perms to the original repo.

This PR represents a first draft of content for a guide on collaborative risk analysis and pre-mortems.

Author bio and profile image pending.

All feedback welcome.

### CODE REVIEW

File: _pages/risk-analysis-and-pre-mortems.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+
+### Practice Psychological Safety
+
+Some people might not have a fully formed or easy to articulate argument for why a risk is significant or a quality attribute is important. Important risks might emerge from someone simply sharing a vague opinion or an intuition that “something around this part of our system worries me…”. Try to encourage an environment where people are able to share ideas freely, even if those ideas are incomplete or if the person is lacking information.
+
+### Observe and Prompt
+
+Some people might find it challenging to feel productive during collaborative exercises, particularly if they introduce new concepts and terminologies. Be prepared to clarify the expectations of the format and each of it's stages and provide examples. If you notice that some attendees are finding it difficult to contribute to a large group you could try applying the 'diverge and merge' technique (see the Nightmare Headline Game format) to encourage more engagement in smaller groups.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+### Share The Workload
+
+Facilitating any type of meeting, agile ceremony or workshop and dealing with the outcomes can be a significant amount of unseen work. Be prepared to share the hosting and admin responsibilities as a team. If you have team members who are looking to gain more experience with hosting collaborative sessions try to understand how you can support them and set them up for success.
+
+### Practice Psychological Safety
+
+Some people might not have a fully formed or easy to articulate argument for why a risk is significant or a quality attribute is important. Important risks might emerge from someone simply sharing a vague opinion or an intuition that “something around this part of our system worries me…”. Try to encourage an environment where people are able to share ideas freely, even if those ideas are incomplete or if the person is lacking information.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+The suggested default limit of 6 quality attributes is really here to encourage prioritisation, reduce the scope of the 'storming' phase and get people thinking about potential trade offs. Some teams might find it challenging to commit to a small subset of quality attributes. Try to remind the team that prioritising some quality attributes doesn't mean that the remaining quality attributes are irrelevant or unimportant.
+
+Depending on the teams context and level of testing maturity some quality attributes might be completely new to some people. This in itself is a valuable signal. Don't ignore it! Instead consider what you can learn from this as a team. Are you exposing a lack of understanding of testing roles? Are there future opportunities here for your testing experts to work more closely with other disciplines?
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+
+### Organisational Context & Culture
+
+A teams perspective on risk might be influenced by the context and culture of an organisation. A small scale startup might have a relaxed attitude towards risk and might intentionally encourage risk taking behaviours. A highly experimental, 'failing fast' culture without formalities might be entirely reasonable in this context. A well established financial organisation, operating in a highly regulated industry and serving millions of customers daily is unlikely to share the same attitude.
+
+Some organisations might associate formal risk management processes with a more waterfall style methodology and have challenges in understanding where risk management fits alongside modern agile software development approaches.
+
+### What are the alternatives?
+
+> “Risk communication and risk management efforts are destined to fail unless there are structured as a two-way process. Each side, expert and public, has something valid to contribute...” Slovic 1987, p.285
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+
+Some organisations might have specific governance roles who are responsible for risk management and defining processes for delivery teams to follow. The visibility of this work and how effectively this role communicates is likely to impact a delivery teams ability to respond to risks. There's potential here for complacency, where teams assume that risk is someone else's responsibility or that if they follow an official governance framework and 'tick the box' they'll be fine.
+
+
+### Organisational Context & Culture
+
+A teams perspective on risk might be influenced by the context and culture of an organisation. A small scale startup might have a relaxed attitude towards risk and might intentionally encourage risk taking behaviours. A highly experimental, 'failing fast' culture without formalities might be entirely reasonable in this context. A well established financial organisation, operating in a highly regulated industry and serving millions of customers daily is unlikely to share the same attitude.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+
+### Relevance
+
+Risks management is ideally an iterative process which is regularly revisited and updated as a project evolves. However in some contexts this process might be reduced to being a one time activity at the very beginning of a project such as a risk workshop. Risks that seemed important to highlight during this initial inception phase might not remain relevant for long.
+
+Some risk management techniques might only provide a limited perspective by focusing only on high level business or project specific risks. These types of risks might not seem particularly relevant or attention grabbing to a team member who is currently focused on day-to-day implementation work and working towards delivery milestones.
+
+Identifying an arbitrary number of risks is unlikely to save a project from potential disaster. As is attempting to create an exhaustive list of all the possible risks threating a project. Some sensible judgement is needed to define a limit on how far to go when identifying potential risks.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+
+Software development is inherently about dealing with uncertainty, unknowns and sometimes surprises.
+
+Any non-trivial software project involves risks. Your team could be developing an internal tool to manage the stock of office kitchen snacks or you could be on a high profile client project migrating a legacy monolithic system to a microservices architecture. Each of these contexts involve risks which could impact your team's delivery and which your team might be accountable for.
+
+Risk management is all about anticipating risks, understanding there likelihood and impact, and identifying actions to mitigate them. But what do we mean by risks?
+
+In the context of a software project, risk tends to be presented as three overlapping categories.
+
+- Project risks - Anything that might threaten the success of your project.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+
+Organise a meeting with a group of people knowledgeable and accountable for the project, product or decision you will be critiquing.
+
+Invite a diverse group of people to the session. To be really effective you'll need a broad range of experience, seniority, skill-sets, roles and perspectives. Don't limit your invitation to just the immediate dev team or 'technical' roles. Include anyone who could be accountable for or impacted by your project being a disaster. There might even be value to inviting domain experts or technical experts who aren't directly involved in your project. If you're struggling for ideas of who to invite, think back to the people you met during your first few weeks on the project. <link to first week guidance>
+
+#### 1. Set The Scene
+
+Briefly review the current state of your project, plan or important decision as a group.
+
+Now set the scene using a short speech to frame peoples thinking. Below is a suggestion from Kahneman. Adapt this to your current context and the time frame of your project.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 9]
```diff
@@ -0,0 +1,401 @@
...
+One of the outcomes of this session will be a collection of exploratory testing charters. If you're unfamiliar with the concepts of exploratory testing and charters it's worth doing some brief upskilling. If there are people in the organisation with experience of these, get them to provide a brief introduction to you.
+
+If software testing isn't your area of expertise, seek out colleagues who are currently working in testing roles or have previous experience in these roles. Ideally you'll want to have at least 1 testing professional attending the session. Their perspective will be valuable in this format and they might be required to take a lead on supporting the group as they draft exploratory testing charters during the final stage.
+
+You might want to create some a template or example exploratory testing charters in a tool of your choice (e.g. Jira, GitHub markdown or a Word document)
+
+If you're conducting an in-person session you'll need:
+- Post it notes
+- Pens
+- A whiteboard/wall
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 10]
```diff
@@ -0,0 +1,401 @@
...
+- Optionally a screen to share content
+
+If you're conducting a remote or hybrid session prepare a digital board using a tool of your choice (e.g. Trello or Miro) before organising the meeting.
+
+#### 1. Set The Scene
+
+Similar to the classic pre-mortem, you'll need to frame people's thinking and encourage them to be imaginative. Start by sharing some real world examples of news reports of well known software failures to frame peoples thinking and provide some inspiration. Even better if you can find a story specific to your project domain.
+
+You might even have stories of your own to share. Where you a customer who was impacted by a high profile production incident? Did you're previous employer make the front page of a major news website for all the wrong reasons? For some inspiration, checkout Gojko Adzic's _Human's Vs Computers_ which is a well referenced [collection of stories of famous bugs](https://gojko.net/books/humans-vs-computers/) and system failures. They range from the hilarious and slightly annoying to the terrifying and deadly.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 11]
```diff
@@ -0,0 +1,401 @@
...
+
+#### 1. Set The Scene
+
+Similar to the classic pre-mortem, you'll need to frame people's thinking and encourage them to be imaginative. Start by sharing some real world examples of news reports of well known software failures to frame peoples thinking and provide some inspiration. Even better if you can find a story specific to your project domain.
+
+You might even have stories of your own to share. Where you a customer who was impacted by a high profile production incident? Did you're previous employer make the front page of a major news website for all the wrong reasons? For some inspiration, checkout Gojko Adzic's _Human's Vs Computers_ which is a well referenced [collection of stories of famous bugs](https://gojko.net/books/humans-vs-computers/) and system failures. They range from the hilarious and slightly annoying to the terrifying and deadly. 
+
+There's a balance here between sparking the teams imagination and accidentally constraining their thought process or biasing them into reproducing previous stories. More experienced teams might not need much in the way of prompting, adjust this to the context of your team.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 12]
```diff
@@ -0,0 +1,401 @@
...
+ - Is it realistic?
+ - Is there a clear impact?
+ - How many unknowns are there?
+ - Is it reasonably within our control?
+
+ The use of intuition in justifying a choice is also perfectly valid here. Someone might not be able to fully articulate why they feel a scenario is important. It might be more of a sense of "There's something here that worries me, but I'm not full sure what...".
+
+#### 4. Discuss potential contributing causes as a group
+
+As with the classic pre-mortem format, the purpose of this stage is to analyse the chosen scenario and break it down into some potential contributing causes. It's also where you'll gain valuable insight into your colleagues different perspectives on risk and quality. There are lots of opportunities for knowledge sharing and clarifying understandings here. It's wise to try to put some time constraints in place and to be mindful of keeping the discussion focused on the chosen scenario.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 13]
```diff
@@ -0,0 +1,401 @@
...
+
+- Ask the group to suggest potential contributing causes to the chosen scenario.
+- Briefly discuss each cause.
+- Ask clarifying questions, but try to avoid limiting people's imagination, biasing their approach or seeking agreement from the group.
+- Record all the proposed contributing causes.
+
+As an Optional addition to this step, ask the group to consider what signals or symptoms they might notice which could suggest this situation is about to occur? This step is attempting to encourage teams to consider how the observability and testability of a system can contribute to their ability to anticipate problems. Some caution is needed here. People without experience of these concepts or who work with systems without a good level of maturity in this area might find this a bit distracting.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 14]
```diff
@@ -0,0 +1,401 @@
...
+
+Make sure you commit to using the charters! It might be helpful at this point to assign ownership of them to a specific person or agree to bring them into the next team planning or prioritisation meeting.
+
+#### 6. Closing the session
+
+> “The main virtue of the pre-mortem is that it legitimizes doubts...” – _Kahneman_
+
+Oh no! You've introduce doubt and potentially reduced the default optimistic view some attendees held. There might be a sudden sense of unease amongst the group due to potential changes in confidence around existing functionality, requirements and project deadlines. You might have even gone far enough to dispelled a long standing illusion that everything in the project is perfectly fine. Managing your business stakeholders expectations is going to be key here and it's another very good reason to invite them along to the session, rather than surprising them with the outcomes later.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 15]
```diff
@@ -0,0 +1,401 @@
...
+- How will the outcomes of your exploratory testing impact the current priorities, workload, velocity and morale of development teams?
+- How will your stakeholders react when you announce that your exploratory testing has discovered missing requirements, unknowns, inconsistencies and potential headline making bugs?
+- What will it take to restore the confidence people held prior to running the session?
+
+### Risk Storming
+
+Risk Storming is a gamified and collaborative test strategy exercise which has been popularised by the Ministry of Testing community. This exercise aims to make discussions about testing and risks accessible to a whole cross functional team and encourages them to take shared responsibility of quality. The game uses the Test Sphere card deck and a simple game board to focus the group as they work through the 3 phases below.
+
+1. The team identifies the 6 most important quality aspects to their product (e.g. performance or accessibility).
+2. The team suggest risks which might impact the quality aspects.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 16]
```diff
@@ -0,0 +1,401 @@
...
+
+If you think your team can handle the potential overheads of new terminologies and game rules this can be a really fun and rewarding exercise. As with all good card games there’s even a TestSphere expansion pack featuring lots of new cards relating to observability!
+
+![riskstorming1]({{ site.baseurl }}/media/risk-analysis-and-pre-mortems/riskstorming1.jpg)
+
+#### Identifying Quality Attributes
+
+During the initial phase of a Risk Storming session the different perspectives of quality within the team will begin to emerge. There might be some people who think of quality purely in terms of functional correctness and if a feature 'works as intended'. Other's might consider functionality as implied and have a preference towards valuing accessibility, security or performance. Neither of these opinions is more or less valuable to the team. If the team have doubts about the importance of a particular quality attribute try to allow some short time-boxed discussions.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 17]
```diff
@@ -0,0 +1,401 @@
...
+
+The suggested default limit of 6 quality attributes is really here to encourage prioritisation, reduce the scope of the 'storming' phase and get people thinking about potential trade offs. Some teams might find it challenging to commit to a small subset of quality attributes. Try to remind the team that prioritising some quality attributes doesn't mean that the remaining quality attributes are irrelevant or unimportant.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 18]
```diff
@@ -0,0 +1,401 @@
...
+
+It can sometimes be useful to separate the format from the real world context of your organisation to allow people to focus on learning. Try substituting a real project or system with something simple which everyone will have some experience of. If you're stuck for ideas [the Ministry of Testing](https://www.ministryoftesting.com/testsphere/riskstorming) suggest performing a Risk Storming session on Iron Man's suit or the Death Star. Remember to be inclusive with your choice of example systems. The idea here is enable the group to learn about the pre-mortem technique and build their confidence for applying it in a real world situation. If some of your colleagues aren't familiar with Star Wars or any of the last 20 years of Marvel universe movies then you could default to something more familiar such as a mobile banking app, cash point machines or restaurant order systems.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: I think you're right, there's an impact Vs intent mismatch here. This statement comes across as dismissive, ironically whilst attempting to make a point on not excluding people.

Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

File: _pages/risk-analysis-and-pre-mortems.md [hunk 19]
```diff
@@ -0,0 +1,401 @@
...
+
+1. The team identifies the 6 most important quality aspects to their product (e.g. performance or accessibility).
+2. The team suggest risks which might impact the quality aspects.
+3. The team attempts to mitigate the risks using known testing techniques, quality heuristics and patterns.
+
+The Ministry of Testing provide a set of [official Risk Storming instructions](https://www.ministryoftesting.com/testsphere/riskstorming), detailed explanations of the 3 phases and free resources to support hosting a session.
+
+As with the classic pre-mortem and the nightmare headline game, encouraging the use of imagination and the sharing of different perspectives are at the core of this format. However, rather than working backwards from hypothetical disaster scenarios, Risk Storming uses the concept of software quality aspects to frame the groups thinking. It also provides opportunities for testing experts to subtly introduce or reiterate concepts such as heuristics. The result is often a collection of more relevant and actionable risks.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Resolved in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/9b3601fc62f44e8c1cbb3556d73e0c89ca182321

**Review by csalt-scottlogic — APPROVED** [review]
I have approved - overall I think it's very good, I have commented on a couple of very minor grammar things.  The only other thing that springs to mind is that you used "e.g." quite a lot, I'd be tempted to replace them with "for example"

File: _pages/risk-analysis-and-pre-mortems.md [hunk 20]
```diff
@@ -0,0 +1,396 @@
...
+
+Pre-mortems encourage us to think about situations that haven't yet occurred and therefore rely on us using our imagination. The importance of this is that it nudges us away from defaulting to our existing models of the world. By limiting the impact of WYSIATI pre-mortems allow us to critique a project or system in a way that enables us to actually start to learn new things about it, rather than just confirming what we already know.
+
+### Countering groupthink
+
+It can be easy for groups to fall into the trap of prioritising conformity over critical thinking. Kahneman explains that as a group approaches a decision the phenomenon of groupthink begins to impact their behaviour. The group's confidence tends to increase and any doubts or alternative opinions tend to be gradually suppressed. Opposers of a decision might tend to be perceived as obstructive to progress, lacking in commitment to the group or simply not understanding the groups goals. Pre-mortems attempt to prevent uncritical optimism from becoming the default of a group by exploring different perspectives and using these to inform better quality decisions.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Fixed in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/1f71b3d9c15f56b85df97ed53d3a8a2f0aaff606

File: _pages/risk-analysis-and-pre-mortems.md [hunk 21]
```diff
@@ -0,0 +1,396 @@
...
+
+### Visibility and Communication
+
+The way in which some traditional risk management approaches are applied might lead to a confusion between identifying and recording risks and actually trying to mitigate and monitor them. This could lead to the assumption that because a risk has been recorded in a risk register document it has been effectively mitigated. In reality, it might just have been safely stored away never to be discussed and reviewed again.
+
+Some organisations might have specific governance roles who are responsible for risk management and defining processes for delivery teams to follow. The visibility of this work and how effectively this role communicates is likely to impact a delivery teams ability to respond to risks. There's potential here for complacency, where teams assume that risk is someone else's responsibility or that if they follow an official governance framework and 'tick the box' they'll be fine.
```
[the remark this answers is not in the document]
Comment (AndrewJanuary) [reply _pages/risk-analysis-and-pre-mortems.md]: Fixed in https://github.com/ScottLogic/practitioners-guides/pull/13/commits/1f71b3d9c15f56b85df97ed53d3a8a2f0aaff606

**Pull request merged**